pydantic/types.py

```diff
@@ -858,6 +863,40 @@ def __get_pydantic_core_schema__(
         def __repr__(self) -> str:
             return 'NaiveDatetime'
 
+    class PastDatetime:
+        @classmethod
+        def __get_pydantic_core_schema__(
+            cls, source: type[Any], handler: GetCoreSchemaHandler
+        ) -> core_schema.CoreSchema:
+            if cls is source:
+                # used directly as a type
+                return core_schema.datetime_schema(now_op='past')
+            else:
+                schema = handler(source)
+                assert schema['type'] == 'datetime'
```

Comment:
What is the purpose of this assertion? If there's a chance it could be hit, it may be worth giving a clearer message of what is going wrong if it _does_ get hit.

Reply:
Actually I don't know how to hit this line. I just keep it because it was in the other types(`AwareDatetime`, `FutureDate`, ...).
@samuelcolvin Do you have any idea here?

Reply:
I guess the assert would fail if you use it as an annotation on something other than `datetime`.

If I'm wrong, we can remove, otherwise, yes a better error would be good.

Reply:
I've added a check that raises an error. added doc and test for error